src/portable/portable.c

```diff
@@ -695,12 +708,12 @@ static int install_chroot_dropin(
         if (!dropin)
                 return -ENOMEM;
 
-        text = strjoin(PORTABLE_DROPIN_MARKER_BEGIN, image_path, PORTABLE_DROPIN_MARKER_END "\n");
+        text = strjoin(PORTABLE_DROPIN_MARKER_BEGIN, m->image ?: image_path, PORTABLE_DROPIN_MARKER_END "\n");
```

Comment:
i don't get this: why is m->image here relevant? i mean, the chroot dropin should always reference the main image, not the extension images. i.e. even if an extension image includes additional service units, which might make sense to honour, they should still be considered to be attached by the main image...

Reply:
Because this is used to identify the image to detach/remove, and if it matches on the image then you can't remove a single attached portable service if the main image is shared.

Rather than "main" image, I think it makes more sense to think of these as "root" image (after all, that's what they are used for). In that sense, you then have a 1:N relationship between "root" and "extensions" - multiple extensions (with units) can use the same root image, but an extension (with units) can be used only once since you cannot have duplicated units.

In concrete terms:

```
portablectl attach --extension app0.raw root.raw
portablectl attach --extension app1.raw root.raw
```

This results in two independent portable services being attached and managed separately, using the same rootfs as the base.

```
portablectl detach app0
```

This detaches the first portable service, but not the second. If the root image identified the portable service once attached, both would be removed.

This is the core of my use case - a flatpak-like "runtime" provided by me, used as the rootfs by multiple application images provided by others.

Reply:
hmm, but what happens if you have two extensions and two root images, and they are all compatible with each other, and each installs a unit file. And then you attach all combinations: root image a with extension x, root image a with extension y, root image a with extensions x *and* y, and root image a with neither. And then the same for b. now if you only track the extension names in the marker string then detaching a will fuck up b's attachment, and reverse. If you only store the root image name however, you fuck things up differently...

maybe we should always generate an "attachment id" from the images used. the id would be a concat of the root image name, plus all extensions in a well defined order. This would then mean, you'd have to specify --extension= fully again when detaching though, but that would be OK, no?

i.e. when you attach ext1.raw, ext3.raw on top of root.raw, we'd generate the id `root.raw+ext1.raw:ext2.raw` or so from it and include that in the marker.

Wouldn't this make sense?

Reply:
That could happen yes, but with the caveat that you'd have to specify different "matches" filters every time, otherwise there would be a clash and the second attachment of the same image would not work.

I'll add the additional ID and push again, hopefully today

Reply:
Ok, the marker now is a colon-separated list of all the images including the root one, in the order in which the are used. I've added WithExtensions detach/inspect methods to make use of this. PTAL!